firebase-firestore/src/main/java/com/google/firebase/firestore/auth/FirebaseAuthCredentialsProvider.java

```diff
@@ -40,7 +44,7 @@
 
   private static final String LOG_TAG = "FirebaseAuthCredentialsProvider";
 
-  private final InternalAuthProvider authProvider;
+  private AtomicReference<Provider<InternalAuthProvider>> internalAuthProviderProviderRef;
```

Comment:
You could also consider storing the `InternalAuthProvider` directly since you seem to need the provider value right away(hence laziness is not an option for you) 
```suggestion
  private final AtomicReference<InternalAuthProvider> internalAuthProviderRef;
```